pennylane_lightning/lightning_qubit.py

```diff
@@ -116,21 +116,23 @@ def apply(self, operations, rotations=None, **kwargs):
             super().apply(operations, rotations=rotations, **kwargs)
             return
 
-        for i, operation in enumerate(operations):  # State preparation is currently done in Python
+        # State preparation is currently done in Python
+        try:  # if operations[0] doesn't exist, that's ok and suppress the error
+            if isinstance(operations[0], QubitStateVector):
+                self._apply_state_vector(operations[0].parameters[0], operations[0].wires)
+                del operations[0]
+            elif isinstance(operations[0], BasisState):
+                self._apply_basis_state(operations[0].parameters[0], operations[0].wires)
+                del operations[0]
+        except IndexError:
+            pass
```

Comment:
`operations` should be a list, so we can check `len`
```suggestion
        if len(operations) > 0:
            if isinstance(operations[0], QubitStateVector):
                self._apply_state_vector(operations[0].parameters[0], operations[0].wires)
                del operations[0]
            elif isinstance(operations[0], BasisState):
                self._apply_basis_state(operations[0].parameters[0], operations[0].wires)
                del operations[0]
```

Reply:
switched to use `if operations:` to check if non-empty.  We don't need to calculate the extensive quantity `len` to check if non-empty.